test/integration/daemonset/daemonset_test.go

```diff
@@ -356,6 +356,42 @@ func validateDaemonSetPodsActive(
 	}
 }
 
+func validateDaemonSetPodsTolerations(
+	podClient corev1client.PodInterface,
+	podInformer cache.SharedIndexInformer,
+	ds *apps.DaemonSet,
+	t *testing.T,
+) {
+	var isMatchToleration bool
+	objects := podInformer.GetIndexer().List()
+	for _, object := range objects {
+		pod := object.(*v1.Pod)
+		ownerReferences := pod.ObjectMeta.OwnerReferences
+		if len(ownerReferences) != 1 {
+			t.Errorf("Pod %s has %d OwnerReferences, expected only 1", pod.Name, len(ownerReferences))
+		}
+		controllerRef := ownerReferences[0]
+		if got, want := controllerRef.Kind, "DaemonSet"; got != want {
+			t.Errorf("controllerRef.Kind = %q, want %q", got, want)
+		}
+		if controllerRef.Controller == nil || *controllerRef.Controller != true {
+			t.Errorf("controllerRef.Controller is not set to true")
+		}
+		for _, dsToleration := range ds.Spec.Template.Spec.Tolerations {
```

Comment:
@atiratree 
Sorry I couldn't come up with good idea.
Since pod tolerations includes tolerations other than daemonset tolerations(ex `node.kubernetes.io/not-ready`), I think we need to check sequentially whether pod tolerations includes each daemonset toleration.

Reply:
hm, I see. The problem with current solution is, that you cannot ask if it is missing some tolerations. I am seeing two possible solutions:
1. pass a test prefix and prefix all your tolerations with it. Then filter all tolerations with this prefix.
2. filter out all tolerations with `node.kubernetes.io` prefix and test no expected tolerations have it.

@mochizuki875 what do you think about these options?

Reply:
> that you cannot ask if it is missing some tolerations

@atiratree 
Sorry, I can't understand current problem...
I understand that the purpose is to ensure that all DaemonSet tolerations are granted to the daemon pods.
Now, we check whether `expectedTolerations` are included in daemon pods or not as below:

```go
		for _, expectedToleration := range expectedTolerations {
			isMatchToleration = false
			for _, podToleration := range pod.Spec.Tolerations {
				if expectedToleration == podToleration {
					isMatchToleration = true
					break
				}
			}
			if !isMatchToleration {
				t.Fatalf("Expected toleration of daemon pod(%s) includes %v, but found %v\n", pod.Name, expectedToleration, pod.Spec.Tolerations)
			}
		}
```

So I think we can check missing tolerations of daemon pod for `expectedTolerations`.
For example, in that case like `expectedTolerations` are not included daemon pods, test will fail and emit log.

```
DaemonSet toleration(expectedTolerations)
  Key: "zone-y", Operator: v1.TolerationOpExists
  Key: "common", Operator: v1.TolerationOpExists

daemon pod toleration
  Key: "common", Operator: v1.TolerationOpExists
```

Example of fail log:
```
daemonset_test.go:389: Expected toleration of daemon pod(foo-hr4wv) includes {zone-y Exists   <nil>}, but found [{common Exists   <nil>} {node.kubernetes.io/not-ready Exists  NoExecute <nil>} {node.kubernetes.io/unreachable Exists  NoExecute <nil>} {node.kubernetes.io/disk-pressure Exists  NoSchedule <nil>} {node.kubernetes.io/memory-pressure Exists  NoSchedule <nil>} {node.kubernetes.io/pid-pressure Exists  NoSchedule <nil>} {node.kubernetes.io/unschedulable Exists  NoSchedule <nil>}]
```

Is this not enough?

Reply:
> that you cannot ask if it is missing some tolerations

@atiratree 
I understand the purpose is: 

1. Ensure that all DaemonSet tolerations are granted to the daemon pods 
2. Emit all missing tolerations as fail log if there are

Now, we can check whether `expectedTolerations` are included in daemon pods or not.
However, we can't emit all missing tolerations as fail log.

For example, in that case like `expectedTolerations` are not included daemon pods, test will fail and only one of missing tolerations is emitted as fail log.
```
DaemonSet toleration(expectedTolerations)
  Key: "zone-y", Operator: v1.TolerationOpExists
  Key: "zone-z", Operator: v1.TolerationOpExists
  Key: "common", Operator: v1.TolerationOpExists

daemon pod toleration
  Key: "common", Operator: v1.TolerationOpExists
```

Example of fail log:
(`{zone-z Exists   <nil>}` also should be emitted, but not)
```
daemonset_test.go:395: Expected toleration of daemon pod(foo-cqmj5) includes [{zone-y Exists   <nil>}], but found [{common Exists   <nil>} {node.kubernetes.io/not-ready Exists  NoExecute <nil>} {node.kubernetes.io/unreachable Exists  NoExecute <nil>} {node.kubernetes.io/disk-pressure Exists  NoSchedule <nil>} {node.kubernetes.io/memory-pressure Exists  NoSchedule <nil>} {node.kubernetes.io/pid-pressure Exists  NoSchedule <nil>} {node.kubernetes.io/unschedulable Exists  NoSchedule <nil>}]
```

So I've fixed to emit all missing tolerations as fail log like that.
```
daemonset_test.go:395: Expected toleration of daemon pod(foo-cqmj5) includes [{zone-y Exists   <nil>} {zone-z Exists   <nil>}], but found [{common Exists   <nil>} {node.kubernetes.io/not-ready Exists  NoExecute <nil>} {node.kubernetes.io/unreachable Exists  NoExecute <nil>} {node.kubernetes.io/disk-pressure Exists  NoSchedule <nil>} {node.kubernetes.io/memory-pressure Exists  NoSchedule <nil>} {node.kubernetes.io/pid-pressure Exists  NoSchedule <nil>} {node.kubernetes.io/unschedulable Exists  NoSchedule <nil>}]
```

How about that?

Reply:
yes that is the main purpose, but we would like to also check that the following example also fails

```
DaemonSet toleration(expectedTolerations)
  Key: "zone-z", Operator: v1.TolerationOpExists
  Key: "common", Operator: v1.TolerationOpExists

daemon pod toleration
  Key: "zone-y", Operator: v1.TolerationOpExists
  Key: "zone-z", Operator: v1.TolerationOpExists
  Key: "common", Operator: v1.TolerationOpExists
```

This has an extra benefit that it will simplify the `validateDaemonSetPodsTolerations` function by using  cmp.diff or DeepEqual

Reply:
@atiratree 
OK, so I've taken your first idea and changed `validateDaemonSetPodsTolerations`.

> 1. pass a test prefix and prefix all your tolerations with it. Then filter all tolerations with this prefix.

Reply:
great, thanks